src/sage/topology/simplicial_complex.py

```diff
@@ -4802,7 +4810,87 @@ def intersection(self, other):
             F = F + [s for s in self.faces()[k] if s in other.faces()[k]]
         return SimplicialComplex(F)
 
+    def bigraded_betti_numbers(self):
+        r"""
+        Return a dictionary of the bigraded Betti numbers of ``self``, 
+        with keys `(-i, 2j)`.
+
+        .. SEEALSO::
+
+            See :meth:`bigraded_betti_number` for more information.
 
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1],[1,2],[1,3],[2,3]])
+            sage: Y = SimplicialComplex([[1,2,3],[1,2,4],[3,5],[4,5]])
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, (-1,4 ): 2, (-1, 6): 1, (-2, 6): 1, (-2, 8): 1}
+            sage: Y.bigraded_betti_numbers()
+            {(0, 0): 1, (-1,4 ): 3, (-2, 6): 1, (-2, 8): 2, (-3, 10): 1}
+        """
+        from sage.homology.homology_group import HomologyGroup
+        if self._bbn is not None:
+            return self._bbn
+        L = self.vertices()
+        n = len(L)
+        B = {}
+        H0 = HomologyGroup(0, ZZ)
+
+        B[(0,0)] = ZZ.one()
+
+        for j in range(n+1):
+            for x in combinations(L, j):
+                S = self.generated_subcomplex(x)
+                H = S.homology()
+                for k in range(j):
+                    if j-k-1 in H and H[j-k-1] != H0:
+                        ind = (-k, 2*j)
+                        if ind not in B:
+                            B[ind] = ZZ.zero()
+                        B[ind] += len(H.get(j-k-1).gens())
+
+        self._bbn = B
+        return self._bbn
+
+    def bigraded_betti_number(self, a, b):
+        if self._bbn is not None:
+            return self._bbn.get((a,b), ZZ.zero())
+        r"""
+        Return the bigraded Betti number indexed in the form `(-i, 2j)`.
+
+        Bigraded Betti number with indices `(-i, 2j)` is defined as a sum of ranks
+        of `(i-j-1)`-th (co)homologies of full subcomplexes with exactly `j` vertices. 
+
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1],[1,2],[2,0],[1,2,3]])
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, ,(-1,4): 1, (-1, 6): 1, (-2, 8): 1}
```

Comment:
```suggestion
            {(0, 0): 1, (-1,4): 1, (-1, 6): 1, (-2, 8): 1}
```
I think there is a typo here.

Can you also add a test checking the different cases:
1. the `(0, 0)` input;
2. an `(a, 2b)` entry that is zero;
3. for `(a, 2b+1)`;
4. The same tests but before setting the cache (see also the comment below).

Reply:
Thank you for noticing. I will add those tests.

Reply:
It should be appropriate now, if I understood you correctly.